app/models/event/answer.rb

```diff
@@ -27,54 +27,21 @@ class Event::Answer < ActiveRecord::Base
   belongs_to :participation
   belongs_to :question
 
+  # TODO
+  attribute :answer, :json
```

Comment:
Das Problem, das hiermit gelöst wird ist folgendes: 

In der bisherigen Implementation wurde die serialization der Antwort unmittelbar beim aufrufen des Setters gemacht. Da dieses Verhalten nun abhängig vom `type` der `Event::Question` ist, ist das nicht mehr möglich.

Der Setter wird unter Umständen vor dem setter der Question aufgerufen, deshalb muss das zu einem späteren Zeitpunkt gemacht werden. Ich habe hierfür `before_validation` gewählt.

Jetzt ists aber so, dass der Default setter für Strings den Array, welcher hier bei Mehrfachantworten übergeben wird, einfach mit `to_s` serialisiert, was später, z.B. in der `before_validation` nicht mehr einfach deserialisiert werden kann.

Mit diesem `attribute :answer, :json` können die Daten aber leicht wieder ausgelesen und weiterverarbeitet werden. 

Hat das zusätzliche Konsequenzen, an welche ich nicht gedacht habe?

Reply:
Allenfalls bei bestehenden Antworten?!

Reply:
Im Bezug auf https://github.com/hitobito/hitobito/pull/2769#discussion_r1735662709 @kronn : 

Ich habe den Kommentar hier am sinnvollsten gefunden